operator_ui/src/pages/Jobs/New.tsx

```diff
@@ -0,0 +1,263 @@
+  TextField,
+  Grid,
+  Card,
+  CardContent,
+  FormControlLabel,
+  FormControl,
+  FormLabel,
+  RadioGroup,
+  Divider,
+  CardHeader,
+  CircularProgress,
+} from '@material-ui/core'
+import {
+  createStyles,
+  withStyles,
+  WithStyles,
+  Theme,
+} from '@material-ui/core/styles'
+import { useLocation, useHistory } from 'react-router-dom'
+import { setPersistJobSpec, getPersistJobSpec } from 'utils/storage'
+
+const jobSpecFormatList = [JobSpecFormats.JSON, JobSpecFormats.TOML]
+export const SELECTED_FORMAT = 'persistSpecFormat'
+
+const styles = (theme: Theme) =>
+  createStyles({
+    card: {
+      padding: theme.spacing.unit,
+      marginBottom: theme.spacing.unit * 3,
+    },
+    loader: {
+      position: 'absolute',
+    },
+  })
+
+const SuccessNotification = ({
+  id,
+}: {
+  id: JobSpec['id'] | OcrJobSpec['id']
+}) => (
+  <>
+    Successfully created job{' '}
+    <BaseLink id="created-job" href={`/jobs/${id}`}>
+      {id}
+    </BaseLink>
+  </>
+)
+
+export function validate({
+  format,
+  value,
+}: {
+  format: JobSpecFormat
+  value: string
+}) {
+  try {
+    if (format === JobSpecFormats.JSON) {
+      JSON.parse(value)
+    } else if (format === JobSpecFormats.TOML) {
+      TOML.parse(value)
+    }
+    return true
+  } catch {
+    return false
+  }
+}
+
+function apiCall({
+  format,
+  value,
+}: {
+  format: JobSpecFormat
+  value: string
+}): Promise<ApiResponse<JobSpec | OcrJobSpec>> {
+  if (format === JobSpecFormats.JSON) {
+    const definition: JobSpecRequest = JSON.parse(value)
+    return api.v2.specs.createJobSpec(definition)
+  }
+
+  if (format === JobSpecFormats.TOML) {
+    const definition: OcrJobSpecRequest = { toml: value }
+    return api.v2.ocrSpecs.createJobSpec(definition)
+  }
+
+  return Promise.reject('Invalid format')
+}
+
+function initialFormat({ query }: { query: string }): JobSpecFormat {
+  const params = new URLSearchParams(query)
+  return (
+    (params.get('format') as JobSpecFormat) ||
+    (storage.get(SELECTED_FORMAT) as JobSpecFormat) ||
+    JobSpecFormats.JSON
+  )
+}
+
+function initialJobSpec(): string {
+  const jobSpec = getPersistJobSpec()
+  try {
+    return JSON.stringify(JSON.parse(jobSpec), null, 4)
+  } catch {
+    return jobSpec || ''
+  }
+}
+
+export const New = ({
+  classes,
+}: {
+  classes: WithStyles<typeof styles>['classes']
+}) => {
+  const location = useLocation()
+  const [format, setFormat] = React.useState<JobSpecFormat>(
+    initialFormat({
+      query: location.search,
+    }),
+  )
+  const [value, setValue] = React.useState<string>(initialJobSpec())
+  const [valid, setValid] = React.useState<boolean>(true)
+  const [loading, setLoading] = React.useState<boolean>(false)
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  React.useEffect(() => {
+    setPersistJobSpec(value)
+    setValid(true)
+  }, [value])
+
+  React.useEffect(() => {
+    storage.set(SELECTED_FORMAT, format)
+    setValid(true)
+    history.push({
```

Comment:
```suggestion
    history.replace({
```
`history.push` adds a history record, so if you start at `/jobs` and click on "New job" you will end up on `/jobs/new?format=json`. Then. if you click back the first time it won't work - the URL will only change to `/jobs/new` without any effect.

Reply:
Good point!